engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/ordering/HasCustomComparatorNode.java

```diff
@@ -30,56 +27,38 @@ public static HasCustomComparatorNode getUncached() {
   }
 
   /**
-   * Returns true if the given atom has a custom comparator, that is a comparator that is different
-   * than the default (internal) ones.
+   * Returns the given atom's comparator if it is a comparator that is different
+   * than the default (internal) one.
    *
    * @param atom Atom for which we check whether it has custom comparator
-   * @return true iff the given atom has a custom comparator
+   * @return {@code null} if the atom has default comparator. Otherwise it returns the real comparator type.
    */
-  public abstract boolean execute(Atom atom);
+  public abstract Type execute(Atom atom);
```

Comment:
If this now returns `Type` then the suffix `Has` doesn't really match the purpose of the call. I would suggest to rename.

Reply:
Right. I didn't want to pollute the diff with needless rename before, but yes, it is desirable to rename the class now: a9787c6b45